### ISSUE TITLE
Native Macro RTE 

### ISSUE THREAD

**Issue by elliottjb7** [umbraco/Umbraco-CMS#6105]
I have just upgraded from 7.4.2 to 7.15.1 and notice that there is still no native Macro RTE property editor. I have created my own for our purposes but feel for those who are unable to do that it must be sorely missed as it's an essential tool when setting up how content is created.

**Comment by nathanwoulfe** [comment]
Is it essential? Personally I avoid macros (DTGE + grid + NC instead), so not entirely sure of the use case?  Example would be awesome.

**Comment by elliottjb7** [comment]
It's absolutely essential for us which is why I created my own. Our website is built using components. Each macro is then inserted into the grid as either a content model or a picker to pick a prewritten component (ie something that's used in multiple locations) - using the macros as content models on a page by page basis allowing us to restrict the content to our design rules. It also makes creating content safe and simple for our non-technical content editors. Of course there's many ways to 'skin a cat' (horrible expression) but this is the most logical for us. What I don't understand is why RTE are excluded from the macros in the first place? Logic would dictate that it's absence is neglect rather than design. Just my thoughts! :)

**Comment by nathanwoulfe** [comment]
@elliottjb7, Interesting - I'd never have considered using macros that extensively. Always find DTGE to give enough flexibility without allowing content editors to create anything too horrible.

Like you said, cats and their skin...

**Comment by nul800sebastiaan** [comment]
Thanks for the suggestion @elliottjb7 ! 

After a few chats at HQ we have some feedback on this:

> What I don't understand is why RTE are excluded from the macros in the first place? Logic would dictate that it's absence is neglect rather than design.

Macro's were always meant to provide a piece of complex functionality that can be configured with parameters. This includes providing some text (for labels and such things) but mostly pickers to use other existing content in the functionality you want to provide. 

Macro's were never meant to be presenting a different way to actually edit content. 

Other than that, the Grid, maybe with DTGE or possibly even Nested Content could be good options for you to explore in your future solutions. Without knowing exactly what you're trying to achieve, we feel that most people get more flexibility and good results out of those combinations. 

In the future, we're working on a new block editor that is a combinations of all we've learned from Macro's, Grid, DTGE and Nested Content (etc.). This should provide you the functionality you need in a very nice way. More info is available in this RFC: https://github.com/umbraco/rfcs/pull/12

In conclusion: it's great that you've found a solution using a custom parameter editor with a rich text editor, great if that works for your purposes. You might even want to make a package out of that as it's probably something other people would like to do as well.

We currently don't plan to offer adding the RTE as a parameter editor for macro's and we look forward to what the new block editor can bring us.
